test/hotspot/jtreg/serviceability/jvmti/Valhalla/FieldAccessModify/libFieldAccessModify.cpp

```diff
@@ -0,0 +1,334 @@
+static bool disableAccessEvent = false;
+
+static void reportError(const char *msg, int err) {
+    printf("%s, error: %d\n", msg, err);
+}
+
+static void printJValue(const char *prefix, JNIEnv *jni_env, char signature_type, jvalue value) {
+    // print new_value to ensure the value is valid
+    // use String.valueOf(...) to get string representation
+    /*
+    Z boolean
+    B byte
+    C char
+    S short
+    I int
+    J long
+    F float
+    D double
+    L fully-qualified-class ;   fully-qualified-class
+    [ type                      type[]
+    */
+    char signature[64] = {};
+    if (signature_type == 'Q' || signature_type == 'L') {
+        sprintf(signature, "(Ljava/lang/Object;)Ljava/lang/String;");
+    } else {
+        sprintf(signature, "(%c)Ljava/lang/String;", signature_type);
+    }
+
+    jclass clsString = jni_env->FindClass("java/lang/String");
+    jmethodID mid = jni_env->GetStaticMethodID(clsString, "valueOf", signature);
+    jstring objJStr = (jstring)jni_env->CallStaticObjectMethodA(clsString, mid, &value);
+
+    const char* objStr = "UNKNOWN";
+    if (objJStr != nullptr) {
+        objStr = jni_env->GetStringUTFChars(objJStr, nullptr);
+    }
+
+    printf("    %s is: '%s'\n", prefix, objStr);
+    fflush(0);
+
+    if (objJStr != nullptr) {
+        jni_env->ReleaseStringUTFChars(objJStr, objStr);
+    }
+}
+
+
+// logs the notification and updates currentTestResult
+static void handleNotification(jvmtiEnv *jvmti, JNIEnv *jni_env,
+    jmethodID method,
+    jobject object,
+    jfieldID field,
+    jclass field_klass,
+    int modified,
+    jlocation location)
+{
+    jvmtiError err;
+    char *name = nullptr;
+    char *signature = nullptr;
+    char *mname = nullptr;
+    char *mgensig = nullptr;
+    jclass methodClass = nullptr;
+    char *csig = nullptr;
+
+    if (testResultObject == nullptr) {
+        // we are out of test
+        return;
+    }
+
+    err = jvmti->GetFieldName(field_klass, field, &name, &signature, nullptr);
+    if (err != JVMTI_ERROR_NONE) {
+        reportError("GetFieldName failed", err);
+        return;
+    }
+
+    err = jvmti->GetMethodName(method, &mname, nullptr, &mgensig);
+    if (err != JVMTI_ERROR_NONE) {
+        reportError("GetMethodName failed", err);
+        return;
+    }
+
+    err = jvmti->GetMethodDeclaringClass(method, &methodClass);
+    if (err != JVMTI_ERROR_NONE) {
+        reportError("GetMethodDeclaringClass failed", err);
+        return;
+    }
+
+    err = jvmti->GetClassSignature(methodClass, &csig, nullptr);
+    if (err != JVMTI_ERROR_NONE) {
+        reportError("GetClassSignature failed", err);
+        return;
+    }
+
+    printf("  \"class: %s method: %s%s\" %s field: \"%s\" (type '%s'), location: %d\n",
+        csig, mname, mgensig, modified ? "modified" : "accessed", name, signature, (int)location);
+
+    // For FieldModification event print current value.
+    // Note: this will cause FieldAccess event.
+    if (modified) {
```

Comment:
It would be cleaner to have modified be declared as a bool, or to re-write the test to modified == 1.

Reply:
> Many of the tests related to the FieldModification event are still not working (they are commented with a KNOWN ISSUE comment) because the withfield bytecode still doesn't support JVMTI FieldModification event. Do you have a CR to track this issue?

Filed JDK-8271355

Reply:
> It would be cleaner to have modified be declared as a bool, or to re-write the test to modified == 1.

C-style detected :)
Will fix it.